Title: agent_kube: requests.SSLError raised on connection using self signed certificates
Class: fix
Compatible: compat
Component: checks
Date: 1725278477
Edition: cre
Level: 1
Version: 2.2.0p33

Newer versions of `requests` don't take `REQUESTS_CA_BUNDLE` into account, resulting in

```
requests.exceptions.SSLError: \
    HTTPSConnectionPool(host='<collector>', port=443): \
        Max retries exceeded with url: \
            /metadata (Caused by SSLError( \
                SSLCertVerificationError(1, \
                    '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: \
                        self signed certificate in certificate chain (_ssl.c:1006)')))
```
being raised if running `agent_kube` against instances using self signed certificates.

This change invokes `session.merge_environment_settings()` to take `REQUESTS_CA_BUNDLE` into
account again.

See
[GitHub: 2807: Use merge_environment_settings method in sessions.send method](https://github.com/psf/requests/issues/2807)
and
[GitHub: 3626: HTTP Proxy with prepared request (honouring env. var.)](https://github.com/psf/requests/issues/3626)

